pkg/v1/tarball/write.go

```diff
@@ -44,78 +44,105 @@ func WriteToFile(p string, tag name.Tag, img v1.Image, wo *WriteOptions) error {
 	tf := tar.NewWriter(w)
 	defer tf.Close()
 
-	// Write the config.
-	cfgName, err := img.ConfigName()
-	if err != nil {
-		return err
-	}
-	cfgBlob, err := img.RawConfigFile()
-	if err != nil {
-		return err
-	}
-	if err := writeTarEntry(tf, cfgName.String(), bytes.NewReader(cfgBlob), int64(len(cfgBlob))); err != nil {
-		return err
-	}
+	var td tarDescriptor
 
-	// Write the layers.
-	layers, err := img.Layers()
-	if err != nil {
-		return err
-	}
-	layerFiles := make([]string, len(layers))
-	for i, l := range layers {
-		d, err := l.Digest()
+	for tag, img := range tagToImage {
+		// Write the config.
+		cfgName, err := img.ConfigName()
 		if err != nil {
 			return err
 		}
-
-		// Munge the file name to appease ancient technology.
-		//
-		// tar assumes anything with a colon is a remote tape drive:
-		// https://www.gnu.org/software/tar/manual/html_section/tar_45.html
-		// Drop the algorithm prefix, e.g. "sha256:"
-		hex := d.Hex
-
-		// gunzip expects certain file extensions:
-		// https://www.gnu.org/software/gzip/manual/html_node/Overview.html
-		layerFiles[i] = fmt.Sprintf("%s.tar.gz", hex)
-
-		r, err := l.Compressed()
+		cfgBlob, err := img.RawConfigFile()
 		if err != nil {
 			return err
 		}
-		blobSize, err := l.Size()
-		if err != nil {
+		if err := writeTarEntry(tf, cfgName.String(), bytes.NewReader(cfgBlob), int64(len(cfgBlob))); err != nil {
 			return err
 		}
 
-		if err := writeTarEntry(tf, layerFiles[i], r, blobSize); err != nil {
+		// Write the layers.
+		layers, err := img.Layers()
+		if err != nil {
 			return err
 		}
-	}
+		layerFiles := make([]string, len(layers))
+		for i, l := range layers {
+			d, err := l.Digest()
+			if err != nil {
+				return err
+			}
+
+			// Munge the file name to appease ancient technology.
+			//
+			// tar assumes anything with a colon is a remote tape drive:
+			// https://www.gnu.org/software/tar/manual/html_section/tar_45.html
+			// Drop the algorithm prefix, e.g. "sha256:"
+			hex := d.Hex
+
+			// gunzip expects certain file extensions:
+			// https://www.gnu.org/software/gzip/manual/html_node/Overview.html
+			layerFiles[i] = fmt.Sprintf("%s.tar.gz", hex)
 
-	// Generate the tar descriptor and write it.
-	td := tarDescriptor{
-		singleImageTarDescriptor{
+			r, err := l.Compressed()
+			if err != nil {
+				return err
+			}
+			blobSize, err := l.Size()
+			if err != nil {
+				return err
+			}
+
+			if err := writeTarEntry(tf, layerFiles[i], r, blobSize); err != nil {
+				return err
+			}
+		}
+
+		// Generate the tar descriptor and write it.
+		sitd := singleImageTarDescriptor{
 			Config:   cfgName.String(),
 			RepoTags: []string{tag.String()},
 			Layers:   layerFiles,
-		},
+		}
+
+		td = append(td, sitd)
 	}
+
 	tdBytes, err := json.Marshal(td)
 	if err != nil {
 		return err
 	}
-	return writeTarEntry(tf, "manifest.json", bytes.NewReader(tdBytes), int64(len(tdBytes)))
+	err = writeTarEntry(tf, "manifest.json", bytes.NewReader(tdBytes), int64(len(tdBytes)))
```

Comment:
why not just `return writeTarEntry(...)`?

Reply:
I was testing something out and forgot to switch this back. It's fixed.